[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)

dzdang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73685

Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_linear_prepack_fp16_numerics
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

dzdang committed:
[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)

Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

[ghstack-poisoned]

jerryzh168 reviewed (approved):
great, thanks!

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

jerryzh168 commented on test/quantization/core/test_quantized_op.py:
```diff
@@ -2880,6 +2880,18 @@ def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
+        input_channels=st.integers(16, 32),
+        output_channels=st.integers(4, 8),
+        exponent=st.integers(0, 8))
+    def test_pack_unpack_fp16(self, input_channels, output_channels, exponent):
```
actually maybe this could have a better name, like test_linear_prepack_fp16_numerics

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang commented:
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34599476).

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_linear_prepack_fp16_numerics
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_linear_prepack_fp16_numerics
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32) (#73685)"


Summary:

A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test Plan:
in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_pack_unpack_fp16
```

Reviewed By: jerryzh168

Pulled By: dzdang

fbshipit-source-id: 5da453e5db4801dde196424282140726c8a4ef1f
(cherry picked from commit ac8910e7feb4eebf677c99f287d48915165a87bf)

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]

dzdang committed:
Update on "[Quant][test] Added test to check if fp16 packing->unpacking yields the same result as to(torch.float16).to(torch.float32)"


Summary: A test was added in test_quantized_op.py that checks whether the fp16 packing and subsequent unpacking of a given fp32 tensor produces the same result as to(torch.float16).to(torch.float32)

Test plan: in pytorch main directory, execute
```
python test/test_quantization.py TestDynamicQuantizedOps.test_linear_prepack_fp16_numerics
```

Differential Revision: [D34599476](https://our.internmc.facebook.com/intern/diff/D34599476)

[ghstack-poisoned]
